File: web/src/app/models/response/subscription.ts
import { SubscriptionStatus } from '../auth';

export interface Subscription {
  plan: string;
  status: SubscriptionStatus;
  /** ISO date the current term began. */
  startedOn: string;
  /** ISO date the current term runs out. */
  renewsOn: string;
  /** Whole days left, floored at zero once the term has run out. */
  daysLeft: number;
  /** Days in the current term, so the profile can draw how much is spent. */
  termDays: number;
}

File: src-tauri/src/models/response/subscription.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Subscription {
    pub plan: String,
    pub status: String,
    pub started_on: String,
    pub renews_on: String,
    pub days_left: i64,
    pub term_days: i64,
}
